clouddriver-sql/src/main/kotlin/com/netflix/spinnaker/clouddriver/sql/event/SqlEventCleanupAgent.kt

```diff
@@ -53,24 +52,17 @@ class SqlEventCleanupAgent(
   override fun run() {
     val duration = Duration.ofDays(properties.maxAggregateAgeDays)
     val cutoff = Instant.now().minus(duration)
-    val limit = dynamicConfigService.getConfig(Int::class.java, EVENT_CLEANUP_CONFIG_KEY, EVENT_CLEANUP_LIMIT)
+    val limit = dynamicConfigService.getConfig(Int::class.java, EVENT_CLEANUP_LIMIT_KEY, EVENT_CLEANUP_LIMIT)
 
     log.info("Deleting aggregates last updated earlier than $cutoff ($duration), max $limit events")
 
     registry.timer(timingId).record {
+      val threshold = Instant.now().minus(duration)
+
       withPool(ConnectionPools.EVENTS.value) {
         val rs = jooq.select(field("aggregate_type"), field("aggregate_id"))
           .from(table("event_aggregates"))
-          .where(
-            // jOOQ always converts timestampDiff into 'INTERVAL DAY TO SECOND' type, which then no longer performs
-            // comparison queries correctly. Doing `timestampDiff().greaterThan()` will cast the result of
-            // timestampDiff to an int, as needed, but then cast the duration into an INTERVAL DAY TO SECOND type,
-            // which will always produce a query returning incorrect results.
-            timestampDiff(currentTimestamp(), field("last_change_timestamp", Timestamp::class.java))
-              .cast(Integer::class.java)
-              // timestampDiff will return `microsecond / 1000`
-              .gt(duration.toMillis().toInt() as Integer)
-          )
+          .where(field("last_change_timestamp").lt(Timestamp(threshold.toEpochMilli())))
```

Comment:
This just seems obvious now. I dunno why I decided to do `timestampdiff` in the first place. 🤦

Reply:
i was wondering about it :)